### PR TITLE
add hw14

### PR DESCRIPTION
**Pull request by IzyNP**

### CODE REVIEW

File: src/main/java/com/izmaylov/hw14/Util.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.izmaylov.hw14;
+
+import java.util.List;
+
+class Util<E> implements Multiplier{
```
Comment (Fabolo12) [inline src/main/java/com/izmaylov/hw14/Util.java]: Should be `implements Multiplier<Number>`

File: src/main/java/com/izmaylov/hw14/Util.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public E get() {
+        return value;
+    }
+
+    @Override
+    public Number[] doubleValueIn(Number[] array) {
+        for (int i = 0; i < array.length; i++) {
+            array[i] = array[i].intValue() * array[i].intValue();
```
Comment (Fabolo12) [inline src/main/java/com/izmaylov/hw14/Util.java]: `intValue()` is an unsafe conversion

File: src/main/java/com/izmaylov/hw14/Util.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        return array;
+    }
+
+
+   public<E extends Number> void sum(E one, E two) { // should work only with numbers
```
Comment (Fabolo12) [inline src/main/java/com/izmaylov/hw14/Util.java]: class parameter overlap

File: src/main/java/com/izmaylov/hw14/Util.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        System.out.println(one.doubleValue() + two.doubleValue());
+    }
+
+
+
+    public <E extends Number> double sumOfArray(List<E> list) {
```
Comment (Fabolo12) [inline src/main/java/com/izmaylov/hw14/Util.java]: a different mechanism must be used

**Review by Fabolo12 — CHANGES_REQUESTED** [review]
8

File: src/main/java/com/izmaylov/hw14/Util.java (hunk 2 above)
Comment (IzyNP) [reply src/main/java/com/izmaylov/hw14/Util.java]: Should I use instanceof here ?

File: src/main/java/com/izmaylov/hw14/Util.java (hunk 4 above)
Comment (IzyNP) [reply src/main/java/com/izmaylov/hw14/Util.java]: ill use here upper bounded wildcart in method singature, is it correct?

**Comment by IzyNP** [comment]
I added the changes in my new commit, can you look at them?

File: src/main/java/com/izmaylov/hw14/Multiplier.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package com.izmaylov.hw14;
+
+public interface Multiplier<T> {
```
Comment (Fabolo12) [inline src/main/java/com/izmaylov/hw14/Multiplier.java:3]: Here should be `<T extends Number>`

**Review by Fabolo12 — APPROVED** [review]
10

**Pull request merged**